libethereum/ClientTest.cpp

```diff
@@ -60,8 +60,9 @@ void ClientTest::setChainParams(string const& _genesis)
         if (params.sealEngineName != "NoProof")
             BOOST_THROW_EXCEPTION(ChainParamsNotNoProof() << errinfo_comment("Provided configuration is not well formatted."));
 
+        setAuthor(params.author);  // Must be called before reopenChain !!!
         reopenChain(params, WithExisting::Kill);
-        setAuthor(params.author); //for some reason author is not being set
+        completeSync();  // set sync state to idle for mining
```

Comment:
Why this? What happens without it?

Reply:
without this. chain state is not being set to idle and test_mineblock does not work

Reply:
This function doesn't do anything related to the sync, I don't think it's the right place for this... Can you maybe move it up the stack, to where `setChainParams()` is called? (`comleteSync()` is a public method)

Reply:
complete*Sync* does not do anything related to Sync ? 
anyway without this function the sync state is not idle and new blocks not being mined in test mode.

Reply:
I meant not `completeSync` but `setChainParams`